linkerd/protocol/h2/src/main/scala/io/buoyant/linkerd/protocol/h2/H2RequestActiveTracer.scala

```diff
@@ -0,0 +1,147 @@
+package io.buoyant.linkerd.protocol.h2
+
+import com.twitter.finagle.buoyant.h2.{Frame, Method, Request, Response, Status, Stream}
+import com.twitter.finagle._
+import com.twitter.finagle.buoyant.h2.Frame.{Data, Trailers}
+import com.twitter.finagle.client.Transporter.EndpointAddr
+import com.twitter.finagle.http.util.StringUtil
+import com.twitter.finagle.naming.buoyant.DstBindingFactory
+import com.twitter.util._
+import io.buoyant.linkerd.RouterContextBuilder
+import io.buoyant.router.RoutingFactory.BaseDtab
+import io.buoyant.router.context.{DstBoundCtx, DstPathCtx}
+import io.buoyant.router.{RouterLabel, RoutingFactory}
+
+/**
+ * Intercepts H2 TRACE requests with a max-forwards and l5d-add-context header.
+ * Returns identification, delegation, and service address information
+ * for a given router as well as any other TRACE responses received from downstream
+ * services. Router information is added to the end of an H2 stream
+ *
+ * A router that receives a tracer request with a max-forwards
+ * greater than 0 decrements the max-forwards header by 1 and
+ * forwards the tracer request to a downstream service that may perform
+ * additional processing of the request.
+ *
+ * @param endpoint the endpoint address of a downstream H2 service identified by a router
+ * @param namers namers used to evaluate a service name
+ * @param dtab base dtab used by the router
+ * @param routerLabel name of router
+ */
+class H2RequestActiveTracer(
+  endpoint: EndpointAddr,
+  namers: DstBindingFactory.Namer,
+  dtab: BaseDtab,
+  routerLabel: String
+) extends SimpleFilter[Request, Response] {
+
+  private[this] val MaxForwards = "max-forwards"
+  private[this] val AddRouterContextHeader = "l5d-add-context"
+
+  private[this] def getRequestTraceResponse(resp: Response, stopwatch: Stopwatch.Elapsed):Future[Response] = {
+    val routerCtxF = RouterContextBuilder(
+      routerLabel,
+      stopwatch,
+      DstPathCtx.current,
+      DstBoundCtx.current,
+      endpoint,
+      namers,
+      dtab)
+
+    routerCtxF.map { ctx =>
+      // format current router context
+      val content = s"""|
```

Comment:
is this just `s"\n${ctx.formatRouterContext.trim}\n"`?

Reply:
What you have doesn't save you from Windows, unfortunately:

```@ """|
                        |banana
                        |""".stripMargin.toSeq
res3: Seq[Char] = WrappedString('\n', 'b', 'a', 'n', 'a', 'n', 'a', '\n')```

Reply:
Darn, well, I will go with `s"\n${ctx.formatRouterContext.trim}\n"` anyways